### ISSUE TITLE
Warning when enum class is types as non-int

### ISSUE THREAD

**Issue by tehKaiN** [grisumbras/enum-flags#8]
Suppose I have something like this:
```
enum class tSphereDrawFlags: uint8_t {
	OK        = 0,
	DANGER    = 1,
	HIGHLIGHT = 2
};
```
I'm getting following warning from GCC:
```
flags/flags.hpp: In instantiation of 'constexpr flags::flags<tSphereDrawFlags> flags::operator&(flags::flags<tSphereDrawFlags>, flags::flags<tSphereDrawFlags>)':
flags/flags.hpp:297:30:   required from 'constexpr typename std::enable_if<flags::is_flags<E>::value, flags::flags<E> >::type operator&(E, E) [with E = tSphereDrawFlags; typename std::enable_if<flags::is_flags<E>::value, flags::flags<E> >::type = flags::flags<tSphereDrawFlags>]'
flags/flags.hpp:147:35: warning: narrowing conversion of '(int)(f1.flags::flags<tSphereDrawFlags>::val_ & f2.flags::flags<tSphereDrawFlags>::val_)' from 'int' to 'flags::flags<tSphereDrawFlags>::impl_type {aka unsigned char}' inside { } [-Wnarrowing]
     return flags{f1.val_ & f2.val_};
                                   ^
```

When I remove `: uint8_t` everything works like a charm. I know I could disable warning -Wnarrowing, but I'd like to keep it enabled.

**Comment by oiffrig** [comment]
You can look at my pull request (#7), it solves the problem.

**Comment by grisumbras** [comment]
Fixed by c3ba7290e3f16ecac580e90f48e5b92173298587.
